crates/runtime/src/instance/allocator.rs

```diff
@@ -77,6 +77,16 @@ pub struct InstanceAllocationRequest<'a> {
     pub wasm_data: *const [u8],
 }
 
+impl<'a> InstanceAllocationRequest<'a> {
+    fn store(&self) -> Option<&mut dyn Store> {
```

Comment:
This'll either need a different signature (taking `&mut self`) or need to be `unsafe` because otherwise I don't think it's safe in Rust to go from `&self` to `&mut T`

Reply:
That being said I'm also wary of this because anything that looks at the store should require that the store is set, so I'd ideally prefer to not have this method and if anyone needs to observe the store then they're on the path that already has the store set and it's safe to assert it's there.